offchain-modules/packages/x/src/ckb/tx-helper/generator.ts

```diff
@@ -1436,3 +1434,13 @@ export function placeholderWitness(senderLockscript: Script): HexString {
       ).serializeJson();
   }
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+export function hashWitness(hasher: any, witness: ArrayBuffer): void {
+  const lengthBuffer = new ArrayBuffer(8);
+  const view = new DataView(lengthBuffer);
+  view.setBigUint64(0, BigInt(new Reader(witness).length()), true);
+
+  hasher.update(lengthBuffer);
+  hasher.update(witness);
+}
```

Comment:
unreuseable function

Reply:
> unreuseable function

This is not only for the code to reuse, but also easy for us human beings to read.